pkgs/applications/networking/gopher/phetch/default.nix

```diff
@@ -0,0 +1,60 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+, pkg-config
+, openssl
+, scdoc
+, which
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "phetch";
+  version = "1.1.0";
+
+  src = fetchFromGitHub {
+    owner = "xvxx";
+    repo = pname;
+    rev = "v${version}";
+
+    sha256 = "sha256-rUcc0OH8M0GhwqgyqTQsti8d9qTu+rnROrgrXKPCWAs=";
+  };
+
+  cargoPatches = [
+    ./10-update-cargo-lock.patch # Fixes building openssl-sys
+  ];
+
+  cargoSha256 = "sha256-zITqk6HfY123p2EEnA0vpXnhopHcUj67T974mSQtM3o=";
+
+  nativeBuildInputs = [ pkg-config scdoc which ];
+  buildInputs = [ openssl ];
+
+  postInstall = ''
+    make manual
+    mkdir -p $out $man/share/man/man1
+    install -m644 doc/phetch.1 $man/share/man/man1
+  '';
+
+  doCheck = false;
```

Comment:
The tests don't even compile, but the program works. You might want to add a comment as to why the tests are disabled.